OAK-9699 Document Elastic Indexes (#497)
apache/jackrabbit-oak: oak-doc/src/site/markdown/query/elastic.md

<!--
   Licensed to the Apache Software Foundation (ASF) under one or more
   contributor license agreements.  See the NOTICE file distributed with
   this work for additional information regarding copyright ownership.
   The ASF licenses this file to You under the Apache License, Version 2.0
   (the "License"); you may not use this file except in compliance with
   the License.  You may obtain a copy of the License at

       http://www.apache.org/licenses/LICENSE-2.0

   Unless required by applicable law or agreed to in writing, software
   distributed under the License is distributed on an "AS IS" BASIS,
   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
   See the License for the specific language governing permissions and
   limitations under the License.
  -->

## Elastic Index

> Oak supports Elasticsearch (Elastic for short) based indexes to support 
> both property constraint and full text constraints. 

Reviewer: ```suggestion
Oak supports Elasticsearch (Elastic for short) based indexes for both property constraint and full text constraints. 
```

> Elastic indexes support similar features than [Lucene][lucene] indexes, 

Reviewer 2: similar features **than** [Lucene][lucene] --> similar features **as** [Lucene][lucene]

however there are differences:

> * The `type` is `elastic`.

Reviewer 2: elasticsearch instead of elastic.

* The index definition needs to be under `/oak:index`.
  Other locations are not supported.

> * The `async` property needs to be set to `async`. 

Reviewer 2: The `async` property needs to be set to `async`.  --> The `async` property needs to be set to `elastic-async`.

  Synchronous, `nrt` or other lanes are not supported.
  Indexes are updated asynchronously.
* `refresh` is ignored.
  Changes take effect immediately after changing them.
  Existing documents in Elasticsearch are not changed.

> * `reindex` is ignored.

Reviewer: what do you mean this is ignored? It's the only way to build an ES index after deploying the index definition directly on the repository (no oak-run).

Author: Ah sorry I thought it's ignored. Will remove it.

  Indexes are automatically built when needed.

>   We recommend to build then using the `oak-run` tool.

Reviewer 2: build them*

>   Indexes are automatically built when needed.
>   We recommend to build then using the `oak-run` tool.

Reviewer: ```suggestion
  Indexes are NOT automatically built when needed. They can be build by setting the `reindex` property to `true` or by using the `oak-run` tool.
  We recommend to build them using the `oak-run` tool.
```

   the License.  You may obtain a copy of the License at

       http://www.apache.org/licenses/LICENSE-2.0

   Unless required by applicable law or agreed to in writing, software
   distributed under the License is distributed on an "AS IS" BASIS,
   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
   See the License for the specific language governing permissions and
   limitations under the License.
  -->

## Elastic Index

Oak supports Elasticsearch (Elastic for short) based indexes for both property constraint and full text constraints. 
Elastic indexes support similar features as [Lucene][lucene] indexes, 
however there are differences:

* The `type` is `elasticsearch`.
* The index definition needs to be under `/oak:index`.
  Other locations are not supported.
* The `async` property needs to be set to `elastic-async`. 
  Synchronous, `nrt` or other lanes are not supported.
  Indexes are updated asynchronously.
* `refresh` is ignored.
  Changes take effect immediately after changing them.
  Existing documents in Elasticsearch are not changed.
* Indexes are NOT automatically built when needed: 

>   They can be build by setting the `reindex` property to `true` or by using the `oak-run` tool.

Reviewer: typo
```suggestion
  They can be built by setting the `reindex` property to `true` or by using the `oak-run` tool.
```
